frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/Footer';
+import Header from '~/components/Header';
+
+import ExerciseSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import {
+  loadSubmissions,
+  searchSubmissions,
+} from '~/redux/exercise-submissions/exercise-submissions.actions';
+import ExerciseSubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import Search from '~/components/Search';
+import { RootState } from '../redux/root.reducer';
+
+import styles from './submissions.module.scss';
+
+interface State {
+  loading: boolean;
+  prevY: number;
+}
+
+class Submissions extends React.Component<ConnectedProps<typeof connector>, State> {
+  private observer: IntersectionObserver;
+
+  private buttonRef = React.createRef<HTMLButtonElement>();
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: props.submissions.submissions === undefined,
+      prevY: 0,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+
+    if (submissions.submissions === undefined) {
+      this.getData();
+    }
+
+    const options = {
+      threshold: 0.7,
+    };
+
+    this.observer = new IntersectionObserver(this.loadMore, options);
+    this.observeNextPage();
+  }
+
+  componentWillUnmount() {
+    this.observer.disconnect();
+  }
+
+  getData = async () => {
+    const { submissions, dispatch } = this.props;
+
+    this.setState({ loading: true });
+
+    try {
+      const newSubmissions = await ExerciseService.getSubmissions(submissions.page, '');
+      dispatch(loadSubmissions(newSubmissions));
+    } catch (err) {
+      SweetAlertService.toast({ type: 'error', text: err });
+    } finally {
+      this.setState({ loading: false });
+    }
+  };
+
+  searchData = async (query: string) => {
+    const { dispatch, submissions } = this.props;
+    this.setState({ loading: true });
+
+    try {
+      const newSubmissions = await ExerciseService.getSubmissions(submissions.page, query);
+      dispatch(searchSubmissions(query, newSubmissions));
+    } catch (err) {
+      SweetAlertService.toast({ type: 'error', text: err });
+    } finally {
+      this.setState({ loading: false });
+    }
+  };
+
+  loadMore = (entries) => {
+    let { prevY } = this.state;
```

Comment:
I'm a little confused about the code here. 

You should only check the `isIntersecting` property on the entry, without any additional math. If that's true, load some more elements.